src/pool/srv_util.c

```diff
@@ -256,35 +256,25 @@ ds_pool_map_tgts_update(struct pool_map *map, struct pool_target_id_list *tgts,
 			D_PRINT("Target (rank %u idx %u) is down.\n",
 				target->ta_comp.co_rank,
 				target->ta_comp.co_index);
-			if (pool_map_node_status_match(dom,
-				PO_COMP_ST_DOWN | PO_COMP_ST_DOWNOUT)) {
-				D_DEBUG(DF_DSMS, "change rank %u to DOWN\n",
-					dom->do_comp.co_rank);
-				dom->do_comp.co_status = PO_COMP_ST_DOWN;
-				dom->do_comp.co_fseq = target->ta_comp.co_fseq;
```

Comment:
could you explain why removing these lines?

Reply:
This was actually a separate bug - the previous code was setting an entire domain (i.e. node) to down if any of its children were down. This was interesting, but was being misinterpreted elsewhere as the node being unavailable, when in fact it was still healthy.